src/util/fields.py

```diff
@@ -153,6 +153,17 @@ def clean_context(context):
         if changed:
             add_to_migration_reports(("ir.filters", id_, name), "Filters/Dashboards")
 
+    fieldnameQuote = '"{}"'.format(fieldname)
+    cr.execute(
+        """
+            UPDATE ir_filters
+              SET sort='[' || ARRAY_TO_STRING(ARRAY_REMOVE(STRING_TO_ARRAY(substring(TRIM(sort),2,LENGTH(TRIM(sort))-2),','),%s),',') || ']'
+             WHERE model_id=%s
+               AND sort ~ %s
+        """,
+        [fieldnameQuote, model, fieldnameQuote],
+    )
+
```

Comment:
Actually, the `sort` column contains a JSON array[^1] (since the start[^2]).
So we can parse it using PostgreSQL JSON functions.
Also, we should handle the "desc" sorts.

```suggestion
    if column_exists(cr, "ir_filters", "sort"):
        cr.execute(
            """
with to_update as (
    select id,
           coalesce(array_to_json(array_agg(sort_item ORDER BY rn) filter (where sort_item not in %s)), '[]') as sort
    from (

        select f.id, s.*
          from ir_filters f
  join lateral jsonb_array_elements_text(f.sort::jsonb)
         with ordinality as s(sort_item, rn)
           on true
        where f.model_id = %s
          and f.sort ~ %s
    ) as matches
   group by id
)
update ir_filters f
   set sort = t.sort
  FROM to_update t
 WHERE f.id = t.id
            """,
            [(fieldname, fieldname + " desc"), model, r"\y{}\y".format(fieldname)],
        )

```



[^1]: in 17.0: https://github.com/odoo/odoo/blob/696389439de6087c643162cf96a3d3f9f4f944a9/addons/web/static/src/search/search_model.js#L2221

[^2]: in 9.0: https://github.com/odoo/odoo/blob/7132a15c38a3b5e5ec232404d21aacfa2ac5874e/addons/web/static/src/js/views/search_menus.js#L234